workers/worker_base.rb

```diff
@@ -25,7 +25,11 @@ def initialize
           # We have the `work` inside a `begin rescue`
           # so that if something fails, the thread still is alive
           begin
-            self.scheduler.schedule { self.work } unless self.should_stop
+            if self.should_stop
+              self.scheduler.shutdown
+            else
+              self.scheduler.schedule { self.work }
```

Comment:
because this is happening in a loop, does this mean `self.scheduler.schedule { self.work }` will be called many times? In this case, we might want to consider removing the thread and just using `self.scheduler`